bitreq/src/response.rs

```diff
@@ -127,6 +127,45 @@ impl Response {
     /// # fn main() -> Result<(), Box<dyn std::error::Error>> { Ok(()) }
     /// ```
     pub fn into_bytes(self) -> Vec<u8> { self.body }
+
+    /// Converts JSON body to a `struct` using Serde.
+    ///
+    /// # Errors
+    ///
+    /// Returns
+    /// [`SerdeJsonError`](enum.Error.html#variant.SerdeJsonError) if
+    /// Serde runs into a problem, or
+    /// [`InvalidUtf8InBody`](enum.Error.html#variant.InvalidUtf8InBody)
+    /// if the body is not UTF-8.
+    ///
+    /// # Example
+    /// In case compiler cannot figure out return type you might need to declare it explicitly:
+    ///
+    /// ```no_run
+    /// use serde_json::Value;
+    ///
+    /// # fn main() -> Result<(), minreq::Error> {
+    /// # let url_to_json_resource = "http://example.org/resource.json";
+    /// // Value could be any type that implements Deserialize!
+    /// let user = minreq::get(url_to_json_resource).send()?.json::<Value>()?;
+    /// println!("User name is '{}'", user["name"]);
+    /// # Ok(())
+    /// # }
+    /// ```
+    #[cfg(feature = "json-using-serde")]
+    pub fn json<'a, T>(&'a self) -> Result<T, Error>
+    where
+        T: serde::de::Deserialize<'a>,
+    {
+        let str = match self.as_str() {
+            Ok(str) => str,
+            Err(_) => return Err(Error::InvalidUtf8InResponse),
+        };
+        match serde_json::from_str(str) {
+            Ok(json) => Ok(json),
+            Err(err) => Err(Error::SerdeJsonError(err)),
```

Comment:
```suggestion
        serde_json::from_str(self.as_str()?)
            .map_err(Error::SerdeJsonError)
```

The existing code returns the incorrect error variant. The docs say `InvalidUtf8InBody` which is what `as_str()` returns.

Reply:
Half right I think. Your suggestion is correct and better but AFAICT it doesn't change the logic. I used the suggestion.